test/testobjects/TestOrchestrations.ts

```diff
@@ -45,6 +45,29 @@ export class TestOrchestrations {
         return output.result;
     });
 
+    public static AnyWithTaskSet: any = df.orchestrator(function*(context: any) {
+        const now = new Date(context.df.currentUtcDateTime).getTime();
+        const deadline = new Date(now + 300000);
+        const timeoutTask = context.df.createTimer(deadline);
+
+        const gate1 = context.df.waitForExternalEvent("A");
+        const gate2 = context.df.waitForExternalEvent("B");
+
+        const winner = yield context.df.Task.any([context.df.Task.all([gate1, gate2]), timeoutTask]);
```

Comment:
I had a hard time understanding the test code (more comments might help :)) but do we have a variation that checks for something like the following?
1. `gate1` completes
2. `timeoutTask` completes
3. `timeoutTask` is the winner

I just want to make sure we're verifying that `all` does not complete early if only a subset of tasks are complete.

Reply:
Sounds good.

I just copied the example directly from the Durable Python folks, but it could definitely use some more descriptors for clarity.

I'm also willing to make a more robust scenario like you describe. Will just have to add a bit of complexity to the history generation.

We do already verify that `df.Task.all` should not complete early with the test `Task.all does not proceed if some parallel tasks have completed`, but it's possible there is some unique behavior when it is a parameter of `df.Task.any()`

Reply:
Actually, looking at the test history, it looks like that scenario is already covered.

If `eventsBeatsTimer` is false in `TestHistories.GetAnyWithTaskSet`, then order of events is

1. "A" completes
2. Timer fires
3. "B" completes.

If it is true, then the order is

1. "A" completes
2. "B" completes"
3. Timer fires.

Reply:
Let me know if the improvements helped here.